xdem/spstats.py

```diff
@@ -1003,22 +1009,273 @@ def patches_method(dh : np.ndarray, mask: np.ndarray[bool], gsd : float, area_si
+             list_fit_fun_label: Optional[list[str]] = None):
+    """
+    Plot empirical variogram, with optionally one or several model fits
+    :param df: dataframe of empirical variogram
+    :param list_fit_fun: list of function fits
+    :param list_fit_fun_label: list of function fits labels
+    :param
+    :return:
+    """
     fig, ax = plt.subplots(1)
     if np.all(np.isnan(df.exp_sigma)):
         ax.scatter(df.bins, df.exp, label='Empirical variogram', color='blue')
     else:
         ax.errorbar(df.bins, df.exp, yerr=df.exp_sigma, label='Empirical variogram (1-sigma s.d)')
 
-    if fit_fun is not None:
-        x = np.linspace(0, np.max(df.bins), 10000)
-        y = fit_fun(x)
+    if list_fit_fun is not None:
+        for i, fit_fun in enumerate(list_fit_fun):
+            x = np.linspace(0, np.max(df.bins), 10000)
+            y = fit_fun(x)
+
+            if list_fit_fun_label is not None:
+                ax.plot(x, y, linestyle='dashed', label=list_fit_fun_label[i], zorder=30)
+            else:
+                ax.plot(x, y, linestyle='dashed', color='black', zorder=30)
 
-        ax.plot(x, y, linestyle='dashed', color='black', label='Model fit', zorder=30)
+        if list_fit_fun_label is None:
+            ax.plot([],[],linestyle='dashed',color='black',label='Model fit')
 
     ax.set_xlabel('Lag (m)')
     ax.set_ylabel(r'Variance [$\mu$ $\pm \sigma$]')
     ax.legend(loc='best')
-    ax.grid()
-    plt.show()
+
+    return ax
+
+def plot_1d_binning(df: pd.DataFrame, var_name: str, statistic_name: str, label_var: Optional[str] = None,
+                    label_statistic: Optional[str] = None, min_count: int = 30):
+    """
+    Plot one statistic and its count along a single binning variable.
+    Input is expected to be formatted as the output of the nd_binning function.
+
+    :param df: output dataframe of nd_binning
+    :param var_name: name of binning variable to plot
+    :param statistic_name: name of statistic of interest to plot
+    :param label_var: label of binning variable
+    :param label_statistic: label of statistic of interest
+    :param min_count: removes statistic values computed with a count inferior to this minimum value
+    """
+
+    if label_var is None:
+        label_var = var_name
+    if label_statistic is None:
+        label_statistic = statistic_name
+
+    # Subsample to 1D and for the variable of interest
+    df_sub = df[np.logical_and(df.nd == 1, np.isfinite(pd.IntervalIndex(df[var_name]).mid))].copy()
+    # Remove statistic calculated in bins with too low count
+    df_sub.loc[df_sub['count']<min_count, statistic_name] = np.nan
+
+    # Need a grid plot to show the sample count and the statistic
+    fig = plt.figure()
+    grid = plt.GridSpec(10, 10, wspace=0.5, hspace=0.5)
+
+    # First, an axis to plot the sample histogram
+    ax0 = fig.add_subplot(grid[:3, :])
+    ax0.set_xticks([])
+
+    # Plot the histogram manually with fill_between
```

Comment:
Why is that needed? Can't you either use Panda's functionality, or plt.bar?

Reply:
It's a bit annoying to deal with `plt.bar` arguments when the width varies for each bar, and other histogram methods need the full distribution (which here has already been binned).
I think this is the simplest approach!